core/cache/model.go

```diff
@@ -577,3 +644,95 @@ func (w *waitUnitChange) close() {
 		close(w.done)
 	}
 }
+
+func (m *Model) updateSummary() {
+	if m.initializer.initializing() {
+		logger.Tracef("skipping update as initializing")
+		return
+	}
+	// This method is only called from within a mutex lock.
```

Comment:
Unfortunately, when this method is called from the cache controller, it is behind the controller's lock and not the model's.
This can get you a concurrent map iter/write error.

Reply:
See https://github.com/juju/juju/pull/11249